packages/react-juce/src/components/ScrollView.ts

```diff
@@ -1,20 +1,94 @@
-import React from 'react';
+import React, { PropsWithChildren } from 'react';
 
 import invariant from 'invariant';
 
 function ScrollViewContentView(props: any) {
   return React.createElement('ScrollViewContentView', props, props.children);
 }
 
-export function ScrollView(props: any) {
+function parseScrollbarColorProp(scrollbarColorProp: string, props: any) {
+  const values = scrollbarColorProp.split(' ');
+
+  invariant(
+    values.length === 2,
+    'scrollbar-color should be a space separated string with two values: "{thumbColor} {trackColor}".'
+  );
+
+  props['scrollbar-thumb-color'] = values[0];
+  props['scrollbar-track-color'] = values[1];
+}
+
+function parseScrollbarWidthProp(scrollBarWidthProp: string | number, props: any) {
+  invariant(
+    typeof scrollBarWidthProp === 'string' || typeof scrollBarWidthProp === 'number',
+    'scrollbar-width should be a string or a number.'
+  )
+
+  if (typeof scrollBarWidthProp === 'string') {
+    switch(scrollBarWidthProp) {
+      // JUCE's look and feel V2 uses 18px as default scrollbar thickness.
+      case 'auto':
+        props['scrollbar-width'] = 18;
+        return;
+      case 'thin':
+        props['scrollbar-width'] = 9;
+        return;
+      case 'none':
+        props['overflow'] = 'hidden';
+        props['scroll-on-drag'] = true;
+        return;
+      default:
+        invariant(
+          false,
+          'scrollbar-width as string should be one of \'auto\', \'thin\' or \'none\'');
+    }
+  }
+
+  props['scrollbar-width'] = scrollBarWidthProp;
+}
+
+function parseOverflowProp(overflowProp: string, props: any) {
+  if (typeof overflowProp !== 'undefined') {
+    props['overflow-x'] = props['overflow-x'] || overflowProp;
+    props['overflow-y'] = props['overflow-y'] || overflowProp;
+  }
+}
+
+export interface ScrollViewProps {
+    'overflow'?: string;
+    'overflow-x'?: string;
+    'overflow-y'?: string;
+    'scrollbar-color'?: string;
+    'scrollbar-width'?: string | number;
+    'scroll-on-drag'?: boolean
+}
+
+export function ScrollView(props: PropsWithChildren<ScrollViewProps>) {
   const child = React.Children.only(props.children);
 
   invariant(
-    child.type === ScrollViewContentView,
+    child && child['type'] === ScrollViewContentView,
     'ScrollView must have only one child, and that child must be a ScrollView.ContentView.'
   );
 
-  return React.createElement('ScrollView', props, child);
+  let {
+    overflow,
+    "scrollbar-color": scrollbarColor,
+    "scrollbar-width": scrollBarWidth,
+    ...other
+  } = props;
+
+  if (typeof scrollbarColor !== 'undefined')
+    parseScrollbarColorProp(scrollbarColor, other);
```

Comment:
I generally try to avoid passing objects into functions and expecting a mutation to that object. I find that pattern creates headaches more often than not.

I'd prefer these functions return their own objects, with their own props subset, and here we could `other = Object.assign(other, parseScrollbarColorProp(scrollbarColor);`

It feels nit picky in this case, but I think this pattern of pure functions is worth being nit picky about :)

Reply:
Yeah totally. I was being lazy there, I remember thinking "ergh" when I wrote that .... Immutable. Object.assign tip make sense. I'm still not thinking in full JS!